modules/templates/util_slice.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package templates
+
+import (
+	"reflect"
+)
+
+type SliceUtils struct{}
+
+func NewSliceUtils() *SliceUtils {
+	return &SliceUtils{}
+}
+
+func (su *SliceUtils) Contains(s, v any) bool {
+	if s == nil {
+		return false
+	}
+	sv := reflect.ValueOf(s)
+	if sv.Kind() != reflect.Slice && sv.Kind() != reflect.Array {
+		panic("invalid type, expected slice or array")
```

Comment:
I think it's not needed because Golang template could report better and more useful error messages to developers, including template name / line number / position number. And the caller won't see the panic.

See the latest commit (by the way, the type %T is also added).